File: frontend/src/lib/testUtils/fakers/challengeRequestFaker.ts
import { faker } from "@faker-js/faker";

import { ChallengeRequest } from "@/lib/apiClient";
import { createFakeMinimalProfile } from "./minimalProfileFaker";
import { createFakePoolKey } from "./poolKeyFaker";

export function createFakeChallengeRequest(
    overrides?: Partial<ChallengeRequest>,
): ChallengeRequest {
    return {
        challengeToken: faker.string.alpha(16),
        requester: createFakeMinimalProfile(),
        recipient: createFakeMinimalProfile(),
        pool: createFakePoolKey(),
        expiresAt: faker.date.future().toISOString(),
        ...overrides,
    };
}

File: backend/Tests/AnarchyChess.Api.TestInfrastructure/Fakes/ChallengeRequestFaker.cs
using AnarchyChess.Api.Challenges.Models;

namespace AnarchyChess.Api.TestInfrastructure.Fakes;

public class ChallengeRequestFaker : RecordFaker<ChallengeRequest>
{
    public ChallengeRequestFaker()
    {
        StrictMode(true);
        RuleFor(x => x.ChallengeToken, f => (ChallengeToken)f.Random.AlphaNumeric(16));
        RuleFor(x => x.Requester, f => new MinimalProfileFaker().Generate());
        RuleFor(x => x.Recipient, f => new MinimalProfileFaker().Generate());
        RuleFor(x => x.Pool, f => new PoolKeyFaker().Generate());
        RuleFor(x => x.ExpiresAt, f => f.Date.Future());
    }
}
